VideoLocker/src/main/java/org/edx/mobile/view/PresenterActivity.java

```diff
@@ -0,0 +1,58 @@
+package org.edx.mobile.view;
+
+import android.os.Bundle;
+import android.support.annotation.CallSuper;
+import android.support.annotation.NonNull;
+import android.support.annotation.Nullable;
+
+import org.edx.mobile.base.BaseFragmentActivity;
+
+public abstract class PresenterActivity<P extends Presenter<V>, V> extends BaseFragmentActivity {
+
+    protected P presenter;
+
+    @NonNull
+    abstract protected P createPresenter();
+
+    @NonNull
+    abstract protected V createView();
+
+    @Override
+    @CallSuper
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        if (null == presenter) {
+            presenter = getLastCustomNonConfigurationInstance();
+            if (null == presenter) {
+                presenter = createPresenter();
+            }
+        }
+    }
+
+    @Override
+    @CallSuper
+    protected void onPostCreate(Bundle savedInstanceState) {
+        super.onPostCreate(savedInstanceState);
+        presenter.attachView(createView());
+    }
+
+    @Override
+    public final P getLastCustomNonConfigurationInstance() {
+        return (P) super.getLastCustomNonConfigurationInstance();
+    }
+
+    @Override
+    public final P onRetainCustomNonConfigurationInstance() {
```

Comment:
Ah, looks like FragmentActivity defines non-deprecated versions of these
methods with the "custom" tag.

On Thu, Mar 17, 2016 at 10:38 PM, Brian Guertin <notifications@github.com>
wrote:

> In VideoLocker/src/main/java/org/edx/mobile/view/PresenterActivity.java
> <https://github.com/edx/edx-app-android/pull/613#discussion_r56545954>:
>
> > +    }
> > +
> > +    @Override
> > +    @CallSuper
> > +    protected void onPostCreate(Bundle savedInstanceState) {
> > +        super.onPostCreate(savedInstanceState);
> > +        presenter.attachView(createView());
> > +    }
> > +
> > +    @Override
> > +    public final P getLastCustomNonConfigurationInstance() {
> > +        return (P) super.getLastCustomNonConfigurationInstance();
> > +    }
> > +
> > +    @Override
> > +    public final P onRetainCustomNonConfigurationInstance() {
>
> No, this method is not deprecated.
>
> However I didn't end up using this class, so happy to delete it for now.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/edx/edx-app-android/pull/613/files/1b79f86bec593c27df395eb26a9eb9586f474ec4#r56545954>
>